test/integration/cds_integration_test.cc

```diff
@@ -194,8 +198,16 @@ TEST_P(CdsIntegrationTest, CdsClusterUpDownUp) {
 // Make sure that clusters won't create new connections on teardown.
 TEST_P(CdsIntegrationTest, CdsClusterTeardownWhileConnecting) {
   initialize();
+  // This line ensures that the ClusterInfoImpl is created already.
   test_server_->waitForCounterGe("cluster_manager.cluster_added", 1);
-  test_server_->waitForCounterExists("cluster.cluster_1.upstream_cx_total");
+  if (this->enableLazyInitStats()) {
+    EXPECT_EQ(test_server_->gauge("cluster.cluster_1.ClusterTrafficStats.inited")->value(), 0);
+    EXPECT_EQ(test_server_->counter("cluster.cluster_1.upstream_cx_total"), nullptr);
+    ASSERT_TRUE(forceCreationOfClusterTrafficStats("cluster_1"));
+    EXPECT_EQ(test_server_->gauge("cluster.cluster_1.ClusterTrafficStats.inited")->value(), 1);
+  } else {
+    EXPECT_NE(test_server_->counter("cluster.cluster_1.upstream_cx_total"), nullptr);
```

Comment:
before making any changes to this test, can you experiment by commenting out the increment/decrement of the "init" gauge in the implementation to ensure these tests fail?

Reply:
are you able to run tests locally? I was assuming you'd do this just in your own workspace.

Reply:
yes, I have run the test in that commit locally and confirmed by the test that stats are gone with the line commented out in the lazy-init.h

Reply:
thanks!